### PR TITLE
machines: scroll noVNC into view

### PR DESCRIPTION
**Pull request by mareklibra**
The noVNC frame and related components are scrolled into view when
Console subtab is opened.

Fixes: https://github.com/cockpit-project/cockpit/issues/6939

### CODE REVIEW

**Comment by mareklibra** [comment]
@garrett , can you please try? I think this behavior meets the requirements:
- do nothing, if noVNC frame is fully visible
- maximalize its visible part otherwise

Control components, including VM name, are always visible.

File: pkg/machines/components/vnc.jsx [hunk 1]
```diff
@@ -24,34 +24,35 @@ import { vmId } from '../helpers.es6';
...
     }
-    const uniqueFrameContainerId = `${vmId(vm.name)}-novnc-frame-container`;
+    const vmIdPrefix = vmId(vm.name);
+    const novncContainerId = `${vmIdPrefix}-novnc-frame-container`;
 
     const encrypt = window.location.protocol === "https:";
     const port = consoleDetail.tlsPort || consoleDetail.port;
 
-    let params = `?host=${consoleDetail.address}&port=${port}&encrypt=${encrypt}&true_color=1&resize=true&containerId=${encodeURIComponent(uniqueFrameContainerId)}`;
+    let params = `?host=${consoleDetail.address}&port=${port}&encrypt=${encrypt}&true_color=1&resize=true&containerId=${encodeURIComponent(vmIdPrefix)}`;
```
Comment (martinpitt) [inline pkg/machines/components/vnc.jsx]: That change is unrelated to the scrolling (I think). Passing a different ID here as URL parameter than the div id seems a bit odd - why is that done?

Comment (mareklibra) [inline pkg/machines/components/vnc.jsx]: In fact, it's no more element id but `prefix` to compose id for different elements within `vnc.html`.
I think we can rename the `containerId` URL param to `containerIdPrefix`.

Before this fix, access to just a single element (former containerId) was needed within vnc.html.
But to have scrolling work, I need to adjust multiple of them to achieve maximum visibility of the iframe _and_ related components (like VM name).

Please let me know, I can rename the URL param.

Comment (martinpitt) [inline pkg/machines/components/vnc.jsx]: Ah, thanks for the explanation! Makes sense, don't bother to rename it.

However, there are three test failures on `testInlineConsole (check_machines.TestMachines)` that  time out waiting for the VNC status to be "normal" - in all three screenshots the console still shows "loading". It *appears* unrelated, but I will retry the tests first, and we should fix them (we might need to wait longer there).

**Comment by martinpitt** [comment]
@mareklibra: That flaky test is hopefully fixed by the "Give sub-VM more time to boot in check-machines" commit in https://github.com/martinpitt/cockpit/tree/test-races . I have some other tests to look at before I propose it for merging.

**Comment by mareklibra** [comment]
@martinpitt , thanks. Seems to be more magic than development :-)

**Comment by stefwalter** [comment]
The tests don't pass here:

```
Traceback (most recent call last):
  File "/build/cockpit/bots/../test/verify/check-machines", line 272, in testInlineConsole
    b.wait(lambda: b.eval_js("document.documentElement.outerHTML.indexOf('noVNC_status_normal') >= 0"))
```

**Comment by martinpitt** [comment]
@stefwalter: I thought I fixed that in #7100; I'll retry the tests.

**Comment by mareklibra** [comment]
Rebased
